pandas/tests/indexes/test_interval.py

```diff
@@ -458,6 +592,30 @@ def test_get_indexer_subintervals(self):
         expected = np.array([0, 0, 0], dtype='intp')
         tm.assert_numpy_array_equal(actual, expected)
 
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_get_indexer_subintervals_updated_behavior(self):
+
+        # a good chance this whole method needs reworking.
+
+        # target = IntervalIndex.from_breaks(np.linspace(0, 2, 5))
```

Comment:
For the case of an index without duplicate values, I think `get_indexer` should semantically work similar to `get_loc`, meaning with Intervals it needs exact matches, but with actual values it gets those that contain the value. It only then does it for multiple ones.

So for a single one, those give the same output:

```
In [61]: index = pd.IntervalIndex.from_breaks(np.linspace(0, 2, 5))

In [62]: index
Out[62]: 
IntervalIndex([(0.0, 0.5], (0.5, 1.0], (1.0, 1.5], (1.5, 2.0]]
              closed='right',
              dtype='interval[float64]')

In [63]: index.get_indexer([pd.Interval(0.0, 0.5, closed='right')])
Out[63]: array([0])

In [64]: index.get_indexer([pd.Interval(0.0, 0.5, closed='right')])[0]
Out[64]: 0

In [69]: index.get_loc(pd.Interval(0.0, 0.5, closed='right'))
Out[69]: 0
```

For multiple ones, you just get them in an array. Only if for `get_loc` you would get a KeyError (so missing Interval), you get a `-1` in `get_indexer`.